### PR TITLE
Re-enable wasi-http for recent wit bindgen changes. Renable tests.

### PR DESCRIPTION
**Pull request by brendandburns**
cc @pchickey

### CODE REVIEW

**Comment by alexcrichton** [comment]
Could [this line](https://github.com/bytecodealliance/wasmtime/blob/eb47240157c094cb8ead186ccb06c414b7a5d8cf/crates/test-programs/build.rs#L13) get flipped as well? Otherwise seems reasonable to me! I'll defer to @pchickey though in case he catches anything I missed.

File: ci/run-tests.sh [hunk 1]
```diff
@@ -2,12 +2,13 @@
 
 cargo test \
     --features "test-programs/test_programs" \
+    --features "test-programs/test_programs_http" \
```
Comment (pchickey) [inline ci/run-tests.sh:5]: We don't actually use this feature anymore, and its an oversight that it is still present in `test-programs/Cargo.toml`. (Once the component model was needed to build the wasi tests, there was no reason to gate the component-model feature on just http, so one feature for building all tests is sufficient.)

**Review by pchickey — COMMENTED** [review]
Thanks! Alex pointed out the line in the test-programs build.rs that needs to be fixed, or better yet just remove that whole const and inline the if's body, since it was just a temporary fix.

File: ci/run-tests.sh (hunk 1 above)
Comment (brendandburns) [reply ci/run-tests.sh]: removed.

**Comment by brendandburns** [comment]
Comments addressed (thanks).

I can't seem to figure out the right magic incantation to run just the HTTP tests locally without running the entire test suite. Command line hint welcome.

**Comment by brendandburns** [comment]
> ```
> [phickey@YF2KCQ775L:crates/test-programs]% cargo test --features test_programs --test wasi-http
> ```

When I run this I see:
```
$ cargo test --features test_programs --test wasi-http
   Compiling test-programs v0.0.0 (/workspaces/wasmtime/crates/test-programs)
error: failed to run custom build command for `test-programs v0.0.0 (/workspaces/wasmtime/crates/test-programs)`

Caused by:
  process didn't exit successfully: `/workspaces/wasmtime/target/debug/build/test-programs-38defddead82d2d5/build-script-build` (exit status: 101)
  --- stdout
  cargo:rerun-if-changed=../wasi-preview1-component-adapter
  wasi reactor adapter: "/workspaces/wasmtime/target/debug/build/test-programs-9cf31c5d68db976e/out/wasi_preview1_component_adapter.reactor.wasm"
  cargo:rerun-if-changed=../wasi-preview1-component-adapter
  wasi command adapter: "/workspaces/wasmtime/target/debug/build/test-programs-9cf31c5d68db976e/out/wasi_preview1_component_adapter.command.wasm"
  cargo:rerun-if-changed=./wasi-tests
  cargo:rerun-if-changed=./command-tests
  cargo:rerun-if-changed=./reactor-tests
  cargo:rerun-if-changed=./wasi-http-tests

  --- stderr
     Compiling wasi-preview1-component-adapter v10.0.0 (/workspaces/wasmtime/crates/wasi-preview1-component-adapter)
      Finished release [optimized] target(s) in 9.12s
     Compiling wasi-preview1-component-adapter v10.0.0 (/workspaces/wasmtime/crates/wasi-preview1-component-adapter)
      Finished release [optimized] target(s) in 2.63s
     Compiling wasi-http-tests v0.0.0 (/workspaces/wasmtime/crates/test-programs/wasi-http-tests)
     Compiling reactor-tests v0.1.0 (/workspaces/wasmtime/crates/test-programs/reactor-tests)
      Finished dev [unoptimized + debuginfo] target(s) in 9.28s
  thread 'main' panicked at 'module "/workspaces/wasmtime/target/debug/build/test-programs-9cf31c5d68db976e/out/wasm32-wasi/debug/reactor_tests.wasm" can be translated to a component: module does not export a function named `cabi_realloc`', crates/test-programs/build.rs:228:10
  note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

But I can see that the tests run correctly in the full CI run.

**Comment by brendandburns** [comment]
I think that this is ready to merge.

**Review by pchickey — APPROVED** [review]

**Comment by pchickey** [comment]
No idea what is going on with that local build failure. Thanks for updating this!

**Comment by brendandburns** [comment]
fwiw, I re-sync'd to head and running tests worked so 🤷

**Pull request merged**